lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -612,6 +612,16 @@ class ConvertAtenAvgPoolOp : public OpConversionPattern<OpTy> {
                                                   strideInts, paddingInts)))
       return rewriter.notifyMatchFailure(op, "invalid pooling parameters");
 
+    // Decode strideInts into strideInts and dilation
+    if (strideInts.size() > Dim) {
```

Comment:
```suggestion
    if (strideInts.size() == 2*Dim) {
```
Since you are extracting up to `2*Dim -1`, I think this would be better.

Reply:
Done